01 Operator Overloading/Operator Overloading.Model/Money.cs

```diff
@@ -110,5 +110,47 @@ private set
                 }
             }
         }
+
+        //------------------Convert Currency Function--------------------//    
+        string[] seperatedData;
+        public Money ConvertCurrency(string targetCurrency)
+        {
+            Parse parseObject = new Parse();
+            double sourceAmount = this.Amount;
+            string sourceCurrency = this.Currency;
+            seperatedData = parseObject.Parsing();
+            double rate1 = CurrencyExchangeFactor(sourceCurrency);
+            double rate2 = CurrencyExchangeFactor(targetCurrency);
+            if (rate2 / rate1 == 0)
+            {
+                throw new Exception("Divide by zero error.");
+            }
+            double amount = (rate2 / rate1) * sourceAmount;
+            Money moneyObject = new Money(amount, targetCurrency);
+            return moneyObject;
+
+        }
+        //---------------Finding Exchange Factor---------------//
+        public double CurrencyExchangeFactor(string currency)
+        {
+            bool isPresent = false;
+            int iterate;
+            for (iterate = 0; iterate < seperatedData.Length; iterate++)
+            {
+                if (seperatedData[iterate].Contains(currency))
+                {
+                    isPresent = true;
+                    break; 
+                }
+            }
+            if (isPresent == false)
+            {
+                throw new System.Exception(Messages.CurrencyNotPresent);
+            }
+            string[] finalSplit = seperatedData[iterate].Split(':');
+            double multiplier;
+            Double.TryParse(finalSplit[1], out multiplier);
```

Comment:
No validation on `finalSplit` length. If try parse fails, nothing is done; where as I had told you to throw error.